### ISSUE TITLE
Output format of default target not respected

### ISSUE THREAD

**Issue by vwkd** [parcel-bundler/parcel#6755]
# 🐛 bug report

Specifying the output format of the target "default" in the `package.json` has no effect. It works fine with any other target name.

EDIT: No, actually only the target name "module". It doesn't seem to work with any other target name. Another bug?

## 🎛 Configuration (.babelrc, package.json, cli command)

`package.json`

```json
"targets": {
    "default": {
      "engines": {
        "outputFormat": "esmodule"
      }
    }
  }
```

## 🤔 Expected Behavior

The file is an ES module.

## 😯 Current Behavior

The file is an IIFE, as produced by the default output format `global`.

## 💁 Possible Solution

A workaround is to use the target name "module". But then it uses that target name as subdirectory of the output directory. To actually use the default output directory of the "default" target, one has to set the `distDir` explicitly. This is cumbersome. It would be easier if we could just overwrite the output format of the default target.

So, in other words, the `package.json` could look like this

```json
"targets": {
    "module": {
      "distDir": "./dist",
      "engines": {
        "outputFormat": "esmodule"
      }
    }
  }
```

EDIT 2: Actually this still uses the output subdirectory "module", so the `distDir` is somehow not applied for the target name "module". Yet another bug?

## 🔦 Context

I want to bundle up some Elm code into an ES module.

## 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| Parcel           | 2.0.0-rc.0 |
| Node             | v16.6.0 |
| npm        | 7.19.1 |
| Operating System | macOS 11.5.2 |

**Comment by mischnic** [comment]
It should be
```
  "targets": {
    "default": {
      "outputFormat": "esmodule"
    }
  }
```

**Comment by vwkd** [comment]
@mischnic Thanks. But this doesn't seem to work. It still outputs as "global".

What does work is changing the target name. I just added it into the "Possible Solutions" section.

**Comment by mischnic** [comment]
Are you sure? The outputFormat config in your "Possible Solutions" example is still wrong

**Comment by vwkd** [comment]
@mischnic Weird. I was sure I read that somewhere.

I tested it more, and the ES module is correctly exported using either

```json
"targets": {
    "module": {
      "outputFormat": "esmodule"
    }
  }
```

or

```json
"targets": {
    "module": {
      "engines": {
        "outputFormat": "esmodule"
      }
    }
  }
```

but neither using

```json
"targets": {
    "default": {
      "outputFormat": "esmodule"
    }
  }
```

nor

```json
"targets": {
    "default": {
      "engines": {
        "outputFormat": "esmodule"
      }
    }
  }
```

EDIT: It doesn't actually work with any target name "foobar" but has to be "module". Another bug?
EDIT2: The output directory `distDir` is actually not applied when using the target name "module". Yet another bug?

**Comment by vwkd** [comment]
Actually, I was wrong. It can't be any target name "foobar". It has to be the target name "module". Is this possibly another bug?

(I edited the description accordingly)

**Comment by vwkd** [comment]
Uh, another possible bug.

```json
"targets": {
    "module": {
      "distDir": "./dist",
      "engines": {
        "outputFormat": "esmodule"
      }
    }
  }
```

works as expected, while

```json
"targets": {
    "module": {
      "distDir": "./dist",
      "outputFormat": "esmodule"
    }
  }
```

doesn't apply the output path `distDir`.

What's going on here? Seems incredibly buggy...

**Comment by mischnic** [comment]
1.
This is just ignored:
```
      "engines": {
        "outputFormat": "esmodule"
      }
```

2. The "module" target has `"outputFormat": "esmodule"` set by default.

3. If you have

```
"module": "build/index.js",
"targets": {
	"module": {
		"distDir": "dist"
	}
}
```
then `build/index.js` will override the `distDir` setting.

**Comment by vwkd** [comment]
@mischnic

All the issues still appear when the "engines" is not present, like for example

```json
"targets": {
    "foobar": {
      "outputFormat": "esmodule"
    }
  }
```

outputs an IIFE.

I count 3 bugs here. I clarified them in my post.

**Comment by mischnic** [comment]
> so the distDir is somehow not applied for the target name "module". Yet another bug?

This is indeed a bug.

---

This is working correctly for me:

```json
  "targets": {
    "default": {
      "outputFormat": "esmodule",
      "isLibrary": true
    }
  },
```

**Comment by thien-do** [comment]
@mischnic I reproduced and filed a bug for the distDir issue: https://github.com/parcel-bundler/parcel/issues/6756 hope it helps :D

**Comment by dotmrjosh** [comment]
There's an object defining [Common targets](https://github.com/parcel-bundler/parcel/blob/375fd68988823186708e46e48f18750ed318f6a5/packages/core/core/src/requests/TargetRequest.js#L57) in [TargetRequst.js](https://github.com/parcel-bundler/parcel/blob/v2/packages/core/core/src/requests/TargetRequest.js) which seems to ignore certain settings, including `distDir`. I don't think it matches exactly the issue going on with #6756 but it might relate. Towards the end of the same file, there's a row statement with a bad docs redirect [here](https://github.com/parcel-bundler/parcel/blob/375fd68988823186708e46e48f18750ed318f6a5/packages/core/core/src/requests/TargetRequest.js#L1245).
When commenting out the `module` key from the `COMMON_TARGETS`, it still doesn't retain the `outputFormat` and switches back to global instead of using esmodules.
